src/include/optimizer/rule_query_rewrite.h

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rule_query_rewrite.h
+//
+// Identification: src/include/optimizer/rule_query_rewrite.h
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "optimizer/rule.h"
+
+#include <memory>
+
+namespace peloton {
+namespace optimizer {
+
+/**
+ * Implements transitive predicates generation for
+ * LogicalGet nodes:
+ *   a = b AND b = c -> a = b AND b = c AND a = c
+ */
+class TransitivePredicatesLogicalGet : public Rule {
+ public:
+  TransitivePredicatesLogicalGet();
+
+  bool Check(std::shared_ptr<OperatorExpression> plan,
+             OptimizeContext *context) const override;
+
+  void Transform(std::shared_ptr<OperatorExpression> input,
+                 std::vector<std::shared_ptr<OperatorExpression>> &transformed,
+                 OptimizeContext *context) const override;
```

Comment:
i think OptimizeContext is usually passed around as a shared_ptr in other parts of the optimizer. you may want to do that for consistency and to gain the benefits of it